java/interfaces_fonctionnelles/en/FUNCTIONAL_INTERFACES.md

```diff
@@ -0,0 +1,241 @@
+
+We now know why functional interfaces are so useful, but we're yet to discover why Java 8 was THE release that really introduced them. Well, as previously stated, Java 8 provides us with a new, shorter and faster syntax to instantiate functional interfaces. Let's take our previous example back:
+
+```java
+Action printAction = new Action() {
+	@Override
+	public void applyActionTo(String str) {
+		System.out.println(str);
+	}
+};
+```
+
+Since, by definition, we are guaranteed that `Action` only has one method to implement, we can now use the following shortcut:
+
+```java
+Action printAction = (str) -> System.out.println(str);
+```
+
+Pretty dope right? Let's go through this example in details.
+
+- As usual, we declare the type of our variable, `Action` in this case, and we assign the variable a name, here `printAction`
+- Since we know that all there is to care about is that `applyActionTo` method, we can define it with a shortcut.
+  - First of all, between the brackets lie the parameters of the method, separated by comas. In our example, there is only one parameter so there is no need for comas. In such a case, it would even be possible to omit the brackets surrounding `str`. You'll notice that since the compiler knows very well the types of the parameters, they're not explicitely stated.
+  - Then, we indicate with an arrow (`->`) that we are going to define the body of the method
+  - Eventually, we do implement the body of the method by simply indicating the instruction to execute. It is possible to have multiple instructions as a body, by encapsulating those instructions in curly braces (`{}`). For instance:
+    ```java
+    Action myAction = (str) -> {
+      System.out.println(str);
+      System.out.println("A String was displayed using a lambda expression!");
+    }
+    
+## Always shorter, always further
+
+In some cases, it is possible to shorten the syntax even more. If your sole instruction as your body is to call a method, and if by any luck the signature of that method happens to match that of the functional interface, then you can reference it directly. Let's take our previous example one more time:
+
+```java
+Action printAction = (str) -> System.out.println(str);
+```
+
+Here, we want a method that asks for a String a returns nothing, it is the very definition of our `applyActionTo` method. And luckily for us, `println` has the same signature! It asks for a `String` and returns nothing. Therefore, we can indicate that our action corresponds to `println`. For the sake of covering all the different cases, we'll create a `static` method as a wrapper to encapsulate in a static context the `println` method (will follow other examples in non-static contexts).
+
+```java
+class MyUtilClass {
+	static void printString(String str) {
+		System.out.println(str);
+	}
+}
+```
+
+We can now assign this method as the value for our functional interface, like this:
+
+```java
+Action printAction = MyUtilClass::printString;

```

Comment:
It's possible to directly do `System.out::println`

Reply:
We talked about it already iirc, yes of course it is possible but it would not be a static context, since `println` is not a static method

Reply:
You wanted to show an example with a static method ?

Reply:
Yes, to cover all cases

Reply:
Okay